### PR TITLE
Fix formatting and clippy warnings

### PR DESCRIPTION
**Pull request by geieredgar**

### CODE REVIEW

File: quinn-h3/src/data.rs [hunk 1]
```diff
@@ -191,7 +191,7 @@ enum WriteFrameState {
 
 impl<F> WriteFrame<F>
 where
-    F: FrameHeader + IntoPayload,
+    F: FrameHeader + Payload,
```
Comment (Ralith) [inline quinn-h3/src/data.rs:194]: What's the story here?

Comment (geieredgar) [reply quinn-h3/src/data.rs:194]: https://rust-lang.github.io/rust-clippy/master/index.html#wrong_self_convention

Should I rename it to `AsPayload` with an `as_payload` function?

Comment (geieredgar) [reply quinn-h3/src/data.rs]: Renamed it.

Comment (Ralith) [reply quinn-h3/src/data.rs]: Ah, skimmed past the trait being renamed below. This seems fine to me as-is.

**Review by Ralith — APPROVED** [review]

**Comment by djc** [comment]
Thanks for fixing these up!

**Pull request merged**